File: etl/models/eleicao.py
from sqlalchemy import Column, BigInteger, Integer, String, Date
from .base import Base

class Eleicao(Base):
    __tablename__ = "eleicao"

    cd_eleicao = Column(BigInteger, primary_key=True)
    nr_turno = Column(Integer, primary_key=True)
    ano_eleicao = Column(Integer, nullable=False)
    cd_tipo_eleicao = Column(Integer)
    nm_tipo_eleicao = Column(String)
    ds_eleicao = Column(String)
    dt_eleicao = Column(Date)

File: src/backend/VoteBem/Entities/Eleicao.cs
namespace VoteBem.Entities
{
    public class Eleicao
    {
        public long CdEleicao { get; private set; }
        public int NrTurno { get; private set; }
        public int AnoEleicao { get; private set; }
        public int? CdTipoEleicao { get; private set; }
        public string? NmTipoEleicao { get; private set; }
        public string? DsEleicao { get; private set; }
        public DateOnly? DtEleicao { get; private set; }

        public ICollection<Coligacao> Coligacoes { get; private set; } = [];
        public ICollection<ResultadoTurno> ResultadosTurno { get; private set; } = [];

        protected Eleicao() { }
    }
}
